server/src/main/java/org/opensearch/search/approximate/ApproximatePointRangeQuery.java

```diff
@@ -0,0 +1,591 @@
+
+                    @Override
+                    public void visit(int docID, byte[] packedValue) {
+                        if (matches(packedValue)) {
+                            visit(docID);
+                        }
+                    }
+
+                    @Override
+                    public void visit(DocIdSetIterator iterator, byte[] packedValue) throws IOException {
+                        if (matches(packedValue)) {
+                            adder.add(iterator);
+                        }
+                    }
+
+                    @Override
+                    public PointValues.Relation compare(byte[] minPackedValue, byte[] maxPackedValue) {
+                        return relate(minPackedValue, maxPackedValue);
+                    }
+                };
+            }
+
+            private boolean checkValidPointValues(PointValues values) throws IOException {
+                if (values == null) {
+                    // No docs in this segment/field indexed any points
+                    return false;
+                }
+
+                if (values.getNumIndexDimensions() != pointRangeQuery.getNumDims()) {
+                    throw new IllegalArgumentException(
+                        "field=\""
+                            + pointRangeQuery.getField()
+                            + "\" was indexed with numIndexDimensions="
+                            + values.getNumIndexDimensions()
+                            + " but this query has numDims="
+                            + pointRangeQuery.getNumDims()
+                    );
+                }
+                if (pointRangeQuery.getBytesPerDim() != values.getBytesPerDimension()) {
+                    throw new IllegalArgumentException(
+                        "field=\""
+                            + pointRangeQuery.getField()
+                            + "\" was indexed with bytesPerDim="
+                            + values.getBytesPerDimension()
+                            + " but this query has bytesPerDim="
+                            + pointRangeQuery.getBytesPerDim()
+                    );
+                }
+                return true;
+            }
+
+            private void intersectLeft(PointValues.PointTree pointTree, PointValues.IntersectVisitor visitor, int count)
+                throws IOException {
+                intersectLeft(visitor, pointTree, count);
+                assert pointTree.moveToParent() == false;
+            }
+
+            private void intersectRight(PointValues.PointTree pointTree, PointValues.IntersectVisitor visitor, int count)
+                throws IOException {
+                intersectRight(visitor, pointTree, count);
+                assert pointTree.moveToParent() == false;
+            }
+
+            // custom intersect visitor to walk the left of the tree
+            private long intersectLeft(PointValues.IntersectVisitor visitor, PointValues.PointTree pointTree, int count)
```

Comment:
Lets please add tests specifically for intersectLeft & Right.

Reply:
added more tests, but im not entirely confident if we're covering them right. would appreciate feedback